project.yaml

```diff
@@ -907,14 +907,43 @@ Layer:
               ('railway_' || (CASE WHEN railway IN ('platform') THEN railway ELSE NULL END)),
               (('aeroway_' || CASE WHEN aeroway IN ('runway', 'taxiway', 'helipad') THEN aeroway ELSE NULL END))
             ) AS feature
-          FROM planet_osm_polygon
-          WHERE highway IN ('residential', 'unclassified', 'pedestrian', 'service', 'footway', 'living_street', 'track', 'path', 'platform', 'services')
-            OR railway IN ('platform')
-            OR aeroway IN ('runway', 'taxiway', 'helipad')
+          FROM (
+          SELECT 
+              way, highway, railway, aeroway, z_order, way_area FROM planet_osm_polygon
+            WHERE highway IN ('residential', 'unclassified', 'pedestrian', 'service', 'footway', 'living_street', 'track', 'path', 'platform', 'services')
+              OR railway IN ('platform')
+              OR aeroway IN ('taxiway', 'helipad')
+          UNION ALL
+          SELECT -- runways with width either tagged or estimated from runway length
+              ST_Buffer(
+                way,
+                0.5*(St_Distance(ST_StartPoint(way), ST_EndPoint(way))/St_Distance_Sphere(ST_Transform(ST_StartPoint(way), 4326), ST_Transform(ST_EndPoint(way), 4326)))*
+                  CASE
+                    WHEN width ~ '^-?\d{1,4}(\.\d+)?$' THEN width::NUMERIC
```

Comment:
> Do you mean a web mercator database or rendering?

Thinking about it, this may not be possible for different rendering projections. If it is possible, it's going to involve `!pixel_width!` and height, as that's the only way we could get any projection-dependent information into the query.

The problem is that to get `width` into rendered projection units, it needs to be multiplied by something for web mercator, and not for local projections where projection units are in meters and are close to true distance. Without a way to get the width into the units of the projection, we're stuck. If we could, we could turn the width into pixels with `sqrt(!pixel_width!*!pixel_height!)`.